### PR TITLE
Ref/feature/menu bar system

### PR DESCRIPTION
**Pull request by MichiGandi**
Add a "menu bar system".

Using the ui builder.

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-34g4tn4 Make menu bar system](https://app.clickup.com/t/34g4tn4)

File: Assets/Scripts/System/MenuBarSystem.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            public List<ContextMenuItem> contextMenuItems;
+
+            public MenuBarItem(string title, List<ContextMenuItem> contextMenuItems)
+            {
+                this.title = title;
+                this.contextMenuItems = contextMenuItems;
+            }
+        }
+
+        public void AddMenuItem(string path, UnityAction onClick, int position = -1)
```
Comment (cblech) [inline Assets/Scripts/System/MenuBarSystem.cs:47]: Please add documentation on how this method works

File: Assets/Scripts/System/MenuBarSystem.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        {
+            _state.AddMenuItem(path, onClick, position);
+            _editorEvents.InvokeUpdateMenuBarEvent();
+        }
+
+        public void DebugCreateExampleMenu()
```
Comment (cblech) [inline Assets/Scripts/System/MenuBarSystem.cs]: Remove testing code.
Or better yet, move it into a unit test

File: Assets/Prefabs/UIDesign/Atoms/Scripts/MenuBarButton.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+using System.Collections;
```
Comment (cblech) [inline Assets/Prefabs/UIDesign/Atoms/Scripts/MenuBarButton.cs]: This script should be placed under `Scripts/Visuals/UiHandlers` with the postfix `Handler`.

**Review by cblech — CHANGES_REQUESTED** [review]

File: Assets/Scripts/System/MenuBarSystem.cs (hunk 1 above)
Comment (MichiGandi) [reply Assets/Scripts/System/MenuBarSystem.cs:47]: Is added.

File: Assets/Scripts/System/MenuBarSystem.cs (hunk 2 above)
Comment (MichiGandi) [reply Assets/Scripts/System/MenuBarSystem.cs]: Core features are tested by unit tests now.

**Review by cblech — APPROVED** [review]

**Pull request merged**